src/v/cloud_storage/types.h

```diff
@@ -60,6 +60,12 @@ enum class manifest_version : int32_t {
     v1 = 1,
 };
 
+enum class tx_range_manifest_version : int32_t {
+    v1 = 1,
+    current_version = v1,
```

Comment:
looks a bit strange to me that current version and compat version are values of enum. I would expect

```
enum class tx_range_manifest_version : int32_t {
    v1 = 1,
};

auto current_version = tx_range_manifest_version:v1;
auto compat_version = tx_range_manifest_version:v1;
```